src/Language/Bitcoin/Script/Descriptors/Syntax.hs

```diff
@@ -43,6 +47,9 @@ data OutputDescriptor
     | -- | An alias for the collection of pk(KEY) and pkh(KEY). If the key is
       -- compressed, it also includes wpkh(KEY) and sh(wpkh(KEY)).
       Combo KeyDescriptor
+    | -- | A P2TR output with the specified key and an optional tree of script
+      -- paths
+      P2TR KeyDescriptor (Maybe TreeDescriptor)
```

Comment:
According to the spec, the only key that can appear here is `XOnlyPubKey`.  So I suggest we make the key field here `XOnlyPubKey`.

Also, we should probably include a warning about stability.  Bitcoin support in Haskell is going through some changes now.  The type of `XOnlyPubKey` is going to change to one exported by the `secp256k1` library bindings and we are shifting development over to https://github.com/haskell-bitcoin .  It doesn't need to block this work, but a comment would be helpful.

Reply:
My understanding of the spec is that key expressions under `tr()` must be interpreted as x-only public keys, but may not necessarily be encoded as x-only public keys in the descriptor: https://github.com/bitcoin/bips/blob/master/bip-0386.mediawiki#modified-key-expression. For example, bitcoin core v23.0 accepts the following expression with compressed public keys:
```bash
bitcoin-cli deriveaddresses "tr(0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798,pk(02c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5))#tmzncdpw"
```
and  secret keys with key origins:
```bash
bitcoin-cli deriveaddresses "tr([00000000]xprv9s21ZrQH143K2JF8RafpqtKiTbsbaxEeUaMnNHsm5o6wCW3z8ySyH4UxFVSfZ8n7ESu7fgir8imbZKLYVBxFPND1pniTZ81vKfd45EHKX73/86'/0'/0'/1/*)#nt6t9skx" [0,10]
```
Should we parse these expressions and translate them into `XOnlyPubKey`s in the AST perhaps with a separate `XOnlyKeyDescriptor` type? 

Source: https://github.com/bitcoin/bips/blob/master/bip-0386.mediawiki#modified-key-expression.

Reply:
I misread the spec!  The way that you have implemented it is just fine.